owtf/managers/plugin.py

```diff
@@ -198,9 +198,30 @@ def plugin_gen_query(session, criteria):
             query = query.filter(Plugin.type.in_(criteria["type"]))
     if criteria.get("group", None):
         if isinstance(criteria["group"], str):
-            query = query.filter(Plugin.group == criteria["group"])
+            plugin_groups_list = Plugin.get_all_plugin_groups(session)
```

Comment:
can you write some tests for this? this is a critical piece of code which might break functionality